backend/corpora/lambdas/api/v1/curation/collections/collection_uuid/assets.py

```diff
@@ -0,0 +1,56 @@
+from flask import g, make_response, jsonify
+
+from backend.corpora.api_server.db import dbconnect
+from backend.corpora.common.entities import Dataset, DatasetAsset
+from backend.corpora.common.utils.http_exceptions import InvalidParametersHTTPException, NotFoundHTTPException
+
+
+@dbconnect
+def get(collection_uuid: str, curator_tag: str = None, dataset_uuid=None):
+    db_session = g.db_session
+    if dataset_uuid:
+        dataset = Dataset.get(db_session, dataset_uuid)
+    elif curator_tag:
+        dataset = Dataset.get_dataset_from_curator_tag(db_session, collection_uuid, curator_tag)
+    else:
+        raise InvalidParametersHTTPException()
+    if not dataset:
+        raise NotFoundHTTPException("Dataset not found.")
+
+    # retrieve the artifact
+    assets = dataset.get_assets()
+    if not assets:
+        raise NotFoundHTTPException("No assets found not found. The dataset may still be processing.")
+
+    asset_list = []
+    error_flag = False
+    for a in assets:
+        asset = DatasetAsset(a)
+        result = dict(file_type=asset.filetype, file_name=asset.filename)
+
+        # Retrieve S3 metadata
+        file_size = asset.get_file_size()
+        if not file_size:
+            result["file_size"] = -1
+            asset_list.append(result)
+            error_flag = True
+            continue
+        else:
+            result["file_size"] = file_size
+
+        # Generate pre-signed URL
+        presigned_url = asset.generate_file_url()
+        if not presigned_url:
+            result["presigned_url"] = "Not Found."
+            asset_list.append(result)
+            error_flag = True
+            continue
+        else:
+            result["presigned_url"] = presigned_url
+        asset_list.append(result)
```

Comment:
It seems a bit strange to use a `continue` directly above in concert with an `else` but then to also have this last line on the parent logic flow. 

Prefer removing `continue` and indenting last line to be under the `else` flow OR remove `else`.

Reply:
^ nit-adjacent 😄